design-documents/service-isolation.md

```diff
@@ -65,195 +35,111 @@ Benefits:
-
-![Isolated Magento services](service-isolation/magento-services.png) 
-
-## Implementation
-To achieve the desired state, two sets of changes are required: platform modifications and service isolation.
-
-### Platform modifications
-
-To support the current ecosystem, the platform and main APIs must stay the same (PHP, Magento Framework).
-
-Work on the platform is required to allow independent service deployment (module separation, application framework, configurable service invokers, backends for frontends, support for split extensions in marketplace). Additional work will  make independently deployed services (cloud native, development environments) manageable.
-
-### Application framework
-
-The following library components are currently implemented as Magento modules. Many modules depend on these library components, which clutters the module dependency graph. Effectively, they will be required for most application services. These library components (or application-agnostic parts of them) should be moved to `Magento\Framework` to reduce clutter in module dependencies:
+### Application decomposition to Services
 
-* Amqp
-* Config
-* Cron
-* Deploy
-* EAV
-* GraphQL
-* Indexer
-* MessageQueue
-* MysqlMq
-* WebAPI
+A service implements distinct business behavior.
+A service consists of one or several modules.
 
-### Configurable service contract invocations
+Communication among services rely on service contracts only.
+Services deployed jointly within one Magento instance use service contracts as calls within a PHP process.
+Services deployed separately within dedicated Magento instances use service contracts as remote calls through REST, SOAP or AMQP APIs.
 
-As of 2.3-develop, Magento service contracts can be called in a process, or through REST, SOAP, and AMQP APIs by remote clients. Remote calls require manual implementation of the client. Magento should be able to invoke service contracts either locally or remotely (if a service whose contract is called is deployed remotely). Two options to implement this are:
+The requirement to communicate through service contracts applies to communication among services only.
+Modules within one service may use direct communication and extendability. 
 
-* Introduce a service contract invoker that should be used by every client to invoke a service contract. For local calls, this invoker would just call a service contract. For remote services, it would make a remote call. This option gives more control over service contract invocation, but requires modification of all current service contract clients.
+### Entry Point Services
 
-* Auto-generated remote call proxies for service contracts that represent remote services. This option does not require code modifications, but it will produce code that is harder to debug.
+Functionality responsible for collaboration with application clients is separated from business logic and located in special endpoint services - BFFs.
```

Comment:
Can we say client (PWA, point of sale, mobile app, etc) specific logic separated from business login?